modules/validations-before-agent-iso-creation.adoc

```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * installing/installing-with-agent/installing-with-agent.adoc
+
+:_content-type: REFERENCE
+[id="validations-before-agent-iso-creation_{context}"]
+= Validation checks before agent.iso is created
+
+The Agent-based installer performs validation checks on user defined yaml files.
+
+=== `install-config.yaml`
+
+* `baremetal`, `vSphere` and `none` platforms are supported.
+* If `none` is used as a platform, the number of control planes should be `1` and the total number of worker replicas should be `0`.
+* `apiVIPs` and `ingressVIPs` must be set for bare metal and vSphere platforms respectively.
+
+=== `agent-config.yaml`
+
+* Mac address has to be defined for each interface.
+* World Wide Name(WWN) extensions are not supported in root device hints.
+* (WWN) vendor extensions are not supported in root device hints.
+* The `role` field in the `host` object must have a value of either `master` or `worker`.
+
+== ZTP manifests
+
+=== `agent-cluster-install.yaml`
+
+* For the `networkType` field, the supported value is `cidr` which is used for IPv6. OpenShift SDN is not supported.
+
+=== `cluster-image-set.yaml`
+
+* The `releaseImage` should have a value of + currentRelease. [question: Can you elaborate on this?]
```

Comment:
This is related to this [PR](https://github.com/openshift/installer/pull/6423) which ensures the releaseImage value used in the cluster-image-set.yaml matches with the registries.conf file contents
- If the ImageContentSource in the install-config doesn't contain the correct releaseImage then this validation error is thrown. This is the case when mirror/registries.conf is generated.
https://github.com/openshift/installer/pull/6423/files#diff-5be39407464453daebd2c1b0fc140dca1556a8963f8c868b68a76f8e5078c92fR78
- If mirror/registries.conf is provided by the user and it contains incorrect releaseImage( not same as the one used in the ClusterImageSet)  then this error is thrown
https://github.com/openshift/installer/pull/6423/files#diff-5be39407464453daebd2c1b0fc140dca1556a8963f8c868b68a76f8e5078c92fR195